### PR TITLE
feat: improve package for /proc/cmdline parsing and management

### PR DESCRIPTION
**Pull request by andrewrynhard**
Signed-off-by: Andrew Rynhard <andrew@andrewrynhard.com>

### CODE REVIEW

**Comment by andrewrynhard** [comment]
This PR makes a number of enhancements to the kernel parameter parser. First of all, this PR allows for multiple instances of an argument to be parsed. For example parsing:

`console=tty0 console=ttyS0`

will now retain _both_ values.

This PR also improvements the performance a bit by reading `/proc/cmdline` once and only once. Previously we depended on parsing `/proc/cmdline` on every lookup for a parameter.

File: internal/app/init/internal/platform/baremetal/baremetal.go [hunk 1]
```diff
@@ -67,25 +68,26 @@ func (b *BareMetal) UserData() (data *userdata.UserData, err error) {
...
-		return err
-	}
-	if err = install.Install(string(cmdlineBytes), data); err != nil {
-		return errors.Wrap(err, "failed to install to bare metal")
+	cmdline := kernel.NewCmdline(kspp.RequiredKSPPKernelParametersString)
+	cmdline.Append("initrd", "/ROOT-A/initramfs.xz")
+	cmdline.Append("console", "tty0")
+	cmdline.Append("console", "ttyS1,115200n8")
+	cmdline.Append("random.trust_cpu", "on")
+	cmdline.Append("talos.platform", "packet")
```
Comment (bradbeam) [inline internal/app/init/internal/platform/baremetal/baremetal.go]: Probably want to drop this one :)

Comment (andrewrynhard) [reply internal/app/init/internal/platform/baremetal/baremetal.go]: whoops :)

File: internal/app/init/internal/platform/baremetal/baremetal.go [hunk 2]
```diff
@@ -67,25 +68,30 @@ func (b *BareMetal) UserData() (data *userdata.UserData, err error) {
...
+// nolint: dupl
 func (b *BareMetal) Install(data *userdata.UserData) (err error) {
-	var cmdlineBytes []byte
-	cmdlineBytes, err = kernel.ReadProcCmdline()
-	if err != nil {
-		return err
-	}
-	if err = install.Install(string(cmdlineBytes), data); err != nil {
-		return errors.Wrap(err, "failed to install to bare metal")
+	cmdline := kernel.NewCmdline(kspp.RequiredKSPPKernelParametersString)
```
Comment (andrewrynhard) [inline internal/app/init/internal/platform/baremetal/baremetal.go]: This is used in three different places, we should probably make this something like `kernel.DefaultParameters()`

**Comment by andrewrynhard** [comment]
```
/ # find /rootfs/boot/
/rootfs/boot/
/rootfs/boot/ROOT-A
/rootfs/boot/ROOT-A/vmlinuz
/rootfs/boot/ROOT-A/initramfs.xz
/rootfs/boot/ROOT-A/include.conf
/rootfs/boot/extlinux
/rootfs/boot/extlinux/extlinux.conf
/rootfs/boot/extlinux/ldlinux.sys
/rootfs/boot/extlinux/ldlinux.c32
/rootfs/boot/installed
/ # cat /rootfs/boot/extlinux/extlinux.conf
DEFAULT ROOT-A
  SAY Talos
INCLUDE /ROOT-A/include.conf
/ # cat /rootfs/boot/ROOT-A/include.conf
LABEL ROOT-A
  KERNEL /ROOT-A/vmlinuz
  INITRD /ROOT-A/initramfs.xz
  APPEND initrd=/ROOT-A/initramfs.xz page_poison=1 slab_nomerge pti=on consoleblank=0 console=ttyS1,115200n8 console=tty0 nvme_core.io_timeout=4294967295 random.trust_cpu=on talos.platform=packet
/ #
```

**Pull request merged**